crates/ruff/src/rules/pylint/rules/return_in_init.rs

```diff
@@ -0,0 +1,47 @@
+use rustpython_parser::ast::{Constant, ExprKind, Stmt, StmtKind};
+
+use ruff_macros::{define_violation, derive_message_formats};
+
+use crate::ast::types::Range;
+use crate::checkers::ast::Checker;
+use crate::registry::Diagnostic;
+use crate::violation::Violation;
+
+use super::yield_in_init::in_dunder_init;
+
+define_violation!(
+    pub struct ReturnInInit;
```

Comment:
Please document the rule (you can take [pylint](https://pylint.readthedocs.io/en/latest/user_guide/messages/error/return-in-init.html) as a start)

See https://github.com/charliermarsh/ruff/issues/2646